contracts/token/BiconomyTokenPaymaster.sol

```diff
@@ -491,7 +491,11 @@ contract BiconomyTokenPaymaster is
             // Transfer full amount to this address. Unused amount will be refunded in postOP
             SafeTransferLib.safeTransferFrom(tokenAddress, userOp.sender, address(this), tokenAmount);
 
-            context = abi.encode(userOp.sender, tokenAddress, tokenAmount, tokenPrice, externalPriceMarkup, userOpHash);
+            // deduct max penalty from the token amount we pass to the postOp
+            // so we don't refund it at postOp
+            // other way to do it is not adding it to the tokenAmount and just charge
+            // tokenAmount + maxPenalty on line 492
+            context = abi.encode(userOp.sender, tokenAddress, tokenAmount-maxPenalty, tokenPrice, externalPriceMarkup, userOpHash);
```

Comment:
maxPenalty is in wei. tokenAmount is number of tokens. how can they be put together in substraction?

Reply:
resolved